### PR TITLE
Refactore source directory and add base code of software engine

### PR DESCRIPTION
**Pull request by szledan**
Signed-off-by: Szilard Ledan <szledan@gmail.com>

### CODE REVIEW

**Comment by kkristof** [comment]
LGTM

File: .travis.yml [hunk 1]
```diff
@@ -92,6 +118,21 @@ matrix:
...
+        - BACKEND=software
+      addons:
+        apt:
+          packages:
+            - freeglut3-dev
+            - libegl1-mesa-dev
+            - libgles2-mesa-dev
+      cache:
+        directories:
+          - tools/frameworks
```
Comment (elecro) [inline .travis.yml]: There's a lot of repetition in this yaml file. Can't we somehow make it into a more simpler form? I think there is the build matrix for Travis which can automatically generate these targets.

Comment (dbatyai) [inline .travis.yml]: This is already on my TODO list.

Comment (elecro) [inline .travis.yml]: Ok cool.

**Comment by elecro** [comment]
lgtm

**Pull request merged**